### PR TITLE
Google oauth2 desktop app

### PR DESCRIPTION
**Pull request by glicht**
## Status
Ready

## Description
Similar to: https://github.com/demisto/content/pull/4967 . But doesn't use oproxy. Instead uses the desktop oauth2 flow provided by google. Still need to configure playbook test.

Make sure to read the long description and follow the setup.

## Must have
- [x] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Technical writer review
There have been modifications to a few parameters in the yaml file since: https://github.com/demisto/content/pull/4967 . Additionally the long description markdown file has changed.
- [ ] [YAML file](https://github.com/demisto/content/edit/google-oauth2-desktop-app/Beta_Integrations/GmailSingleUser/GmailSingleUser.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F5109)
- [ ] [Long Description](https://github.com/demisto/content/edit/google-oauth2-desktop-app/Beta_Integrations/GmailSingleUser/GmailSingleUser_description.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F5109)

### CODE REVIEW

**Comment by glicht** [comment]
@kirbles19 please review also the following message we display when running `!gmail-auth-link`:

## Gmail Auth Link
Please follow the following **[link]({link})**.

After Completing the authentication process, copy the received code
to the **Auth Code** configuration parameter of the integration instance.
Save the integration instance and then run *!gmail-auth-test* to test that
the authentication is properly set.

**Comment by kirbles19** [comment]
@glicht Done.

**Comment by glicht** [comment]
@yaakovi added to **test_integration.py** support for `validate_test`. When set to false will not run the "Test" button on the instance. So we can test the Gmail Single User with a playbook.

File: Beta_Integrations/GmailSingleUser/GmailSingleUser.yml [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  required: false
+  type: 0
+- display: Trust any certificate (not secure)
+  name: insecure
+  required: false
+  type: 8
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
```
Comment (yaakovi) [inline Beta_Integrations/GmailSingleUser/GmailSingleUser.yml]: should be at the end of the parameter list

File: Beta_Integrations/GmailSingleUser/GmailSingleUser.yml [hunk 2]
```diff
@@ -0,0 +1,211 @@
+category: Messaging
```
Comment (yaakovi) [inline Beta_Integrations/GmailSingleUser/GmailSingleUser.yml]: should this be "Email Gateway"?
@Arsenikr WDYT?

File: Beta_Integrations/GmailSingleUser/GmailSingleUser.py [hunk 3]
```diff
@@ -0,0 +1,947 @@
...
+        # Log exceptions
+    except Exception as e:
+        import traceback
+        if command == 'fetch-incidents':
+            LOG(traceback.format_exc())
+            LOG.print_log()
+            raise
+
+        else:
+            return_error('GMAIL: {}'.format(str(e)), traceback.format_exc())
```
Comment (yaakovi) [inline Beta_Integrations/GmailSingleUser/GmailSingleUser.py]: we can use `return_error` on both cases as now it supports `fetch-incidents`

File: Beta_Integrations/GmailSingleUser/GmailSingleUser.py [hunk 4]
```diff
@@ -0,0 +1,947 @@
...
+        'Type': entryTypes['note'],
+        'Contents': markdown,
+        'ReadableContentsFormat': formats['markdown'],
+        'HumanReadable': markdown,
+    }
+
+
+def auth_test_command(client):
+    client.search('me', '', '', '', '', '', '', '', '',
+                  '', [], 10, '', False, False)
```
Comment (yaakovi) [inline Beta_Integrations/GmailSingleUser/GmailSingleUser.py]: most of the values here are the default ones, why pass them?

File: Beta_Integrations/GmailSingleUser/GmailSingleUser.yml (hunk 2 above)
Comment (glicht) [reply Beta_Integrations/GmailSingleUser/GmailSingleUser.yml]: Message is ok. This is also where we have mail listener/sender.

File: Beta_Integrations/GmailSingleUser/GmailSingleUser.py (hunk 4 above)
Comment (glicht) [reply Beta_Integrations/GmailSingleUser/GmailSingleUser.py]: Let's just leave this for now.

**Pull request merged**